apis/telemetry/v1alpha1/logpipeline_conversion.go

```diff
@@ -17,255 +25,210 @@ import (
 // - input.runtime namespaces and containers are now pointers in v1beta1, requiring nil checks during conversion.
 // Additionally, changes were done in shared types which are documented in the related file.
 
+// dataAnnotation is the annotation that conversion webhook can use to retain the data in case of down-conversion from the hub
+const dataAnnotation = "telemetry.kyma-project.io/conversion-data"
+
 var errSrcTypeUnsupportedLogPipeline = errors.New("source type is not LogPipeline v1alpha1")
 var errDstTypeUnsupportedLogPipeline = errors.New("destination type is not LogPipeline v1beta1")
 
-// ConvertTo converts this LogPipeline to the Hub version (v1alpha1 -> v1beta1).
 func (lp *LogPipeline) ConvertTo(dstRaw conversion.Hub) error {
+	logf.FromContext(context.Background()).V(0).Info("converting from v1alpha1 to v1beta1")
+
 	src := lp
 
 	dst, ok := dstRaw.(*telemetryv1beta1.LogPipeline)
 	if !ok {
 		return errDstTypeUnsupportedLogPipeline
 	}
 
-	// Copy metadata
-	dst.ObjectMeta = src.ObjectMeta
+	// Call the conversion-gen generated function
+	if err := Convert_v1alpha1_LogPipeline_To_v1beta1_LogPipeline(src, dst, nil); err != nil {
+		return err
+	}
 
-	// Copy Input fields
-	dst.Spec.Input = telemetryv1beta1.LogPipelineInput{}
-	dst.Spec.Input.Runtime = convertApplicationToBeta(src.Spec.Input.Application)
-	dst.Spec.Input.OTLP = convertOTLPInputToBeta(src.Spec.Input.OTLP)
+	return marshalData(src, dst)
```

Comment:
```suggestion
    // It is necessary to store the v1alpha1 LogPipeline object as an annotation in the v1beta1 LogPipeline object, because the "system" field in the NamespaceSelector in v1alpha1 doesn't have a corresponding field in v1beta1
	return marshalData(src, dst)
```